apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -277,38 +287,68 @@ type EmbeddedObjectMeta struct {
 // TODO Implement Stringer interface. It will helpful for debugging and testing.
 type CassandraConfig struct {
 	// +optional
-	CassandraYaml *CassandraYaml `json:"cassandraYaml,omitempty"`
+	CassandraYaml CassandraYaml `json:"cassandraYaml,omitempty"`
 
 	// +optional
-	JvmOptions *JvmOptions `json:"jvmOptions,omitempty"`
+	JvmOptions JvmOptions `json:"jvmOptions,omitempty"`
 }
 
-type Auth struct {
+type CassandraAuth struct {
+
+	// The reference to the superuser secret to use for Cassandra. If nil and auth is enabled, a default secret will
+	// be generated with a random password. If nil and auth is disabled, no secret will be generated.
 	// +optional
-	Enabled bool `json:"enabled,omitempty"`
+	SuperuserSecretRef corev1.LocalObjectReference `json:"superuserSecretRef,omitempty"`
 
+	// The cache validity period in milliseconds to apply to all authentication caches in Cassandra (roles, permissions
+	// and credentials). If defined, this value is automatically applied to the following cassandra.yaml settings:
+	// - roles_validity_in_ms
+	// - permissions_validity_in_ms
+	// - credentials_validity_in_ms
+	// Setting this value to 0 disables caching entirely.
 	// +optional
 	CacheValidityPeriodMillis *int64 `json:"cacheValidityPeriodMillis,omitempty"`
```

Comment:
@jsanda I kept these 2 cache settings around since you had them initially, but honestly they force us to pass the `CassandraAuth` struct around which is a bit cumbersome. Do you see a lot of value in these settings?

Reply:
Especially keeping in mind that we already have the cluster vs dc template overrides that could be used to specify default settings.

Reply:
I'd get rid of them.